### PR TITLE
#33 - Display GitHub input when the user is Graduate - Sabella

### LINKED ISSUE #33: [Backend] Add GitHub Link Input for Graduate User Type in Sign Up Form
Modify the SignUp component to include an additional input field for the GitHub link when the user selects "Graduate" as their user type. This field should only appear if "Graduate" is selected and should be required.

#26

### PR DESCRIPTION
**Pull request by Sabella-8**
## Naming Rules

Name your PR like this: ISSUENUMBER-TITLE-YOURNAME

## Description

Write a summary of your change. Explain why you have made it.

## Related to

Make sure you include the issue number with a hash sign # so Github can link this PR to the right issue, like this:

Fixes #1

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have carefully reviewed my own code
- [ ] I have commented my code
- [ ] I have updated any documentation

### CODE REVIEW

File: client/src/pages/SignUp.js [hunk 1]
```diff
@@ -8,6 +8,7 @@ const SignUp = () => {
 	const [userType, setUserType] = useState("graduate");
 	const [message, setMessage] = useState("");
 	const [isGraduate, setIsGraduate] = useState(true);
+	const [userGithub, setUserGithub] = useState("");
```
Comment (taiwo2) [inline client/src/pages/SignUp.js:11]: Nice job done

File: server/api.js [hunk 2]
```diff
@@ -13,11 +13,8 @@ router.get("/", (_, res) => {
 	res.json({ message: "Hello, world!" });
 });
 
-
-
-
-router.post("/users",  async (req, res) => {
-	const { username, passwordHash, userType } = req.body;
+router.post("/users", async (req, res) => {
```
Comment (taiwo2) [inline server/api.js:16]: this should be signup  by name not users

**Review by taiwo2 — COMMENTED** [review]
Kindly implement the first for the Readme and pinned project as discussed. in another pull request

File: server/api.js [hunk 3]
```diff
@@ -13,11 +13,8 @@ router.get("/", (_, res) => {
...
+	const { username, passwordHash, userType, userGithub } = req.body;
```
Comment (taiwo2) [inline server/api.js:17]: Kindly use bcrypt to hash the password from the backend, not the frontend

**Comment by Sabella-8** [comment]
Hello, @taiwo2, thank you for your comment. However, the comment you provided is related to the previous work, which I did not do. I only added the GitHub input when users choose "graduate" in the options.

**Pull request merged**